### PR TITLE
End of Candidates

### LINKED ISSUE #118: End of Candidates
The current WebRTC task spec does not allow to send `null` to indicate that no more candidates will be sent (until an ICE restart occurs).

Suggesting to allow `null` in the *candidate* field of the 'candidates' message.

### PR DESCRIPTION
**Pull request by lgrahl**
As discussed in #118, closes #118.

### CODE REVIEW

**Review by dbrgn — APPROVED** [review]

**Comment by dbrgn** [comment]
As far as I can see, this requires no updates in the saltyrtc-task-webrtc-js library. The Java implementation just needs to update some annotations.

Note by the way that the candidate gathered by WebRTC is `null`, while we are sending `{candidate: null, sdpMid: null, sdpMLineIndex: null}`. That's intended, right? It means that the `event.candidate` field of the browser event can't just be passed on to SaltyRTC directly, instead the `event.candidate` field needs to be inspected to generate the empty candidate data for SaltyRTC in case the field is `null`.

```js
if (e.candidate === null) {
    sendCandidate({
        candidate: null,
        sdpMid: null,
        sdpMLineIndex: null,
    });
} else {
    sendCandidate({
        candidate: e.candidate.candidate,
        sdpMid: e.candidate.sdpMid,
        sdpMLineIndex: e.candidate.sdpMLineIndex,
    });
}
```

In case we could make the entire field nullable, it would be:

```js
sendCandidate(e.candidate);
```

On the other hand that could be error prone, in case `e.candidate` includes more fields that we don't accept.

**Comment by lgrahl** [comment]
Oh, the whole thing is `null`, you're right. I'll update the PR. I did not intend to make this incompatible with the API.

Could you rephrase your last sentence? I don't understand what would be error prone. Edit: I think I understand now. You mean, that there could be further fields, right? That could happen at some point, so it's probably a good idea to explicitly set the fields when sending as it's required to validate the fields upon receiving.

```js
if (e.candidate === null) {
    sendCandidate(null);
} else {
    sendCandidate({
        candidate: e.candidate.candidate,
        sdpMid: e.candidate.sdpMid,
        sdpMLineIndex: e.candidate.sdpMLineIndex,
    });
}
```

**Comment by dbrgn** [comment]
> Edit: I think I understand now. You mean, that there could be further fields, right?

Exactly :) Yes, although it's convenient, it's probably better to always use explicit field assignment.

**Review by dbrgn — APPROVED** [review]
Looks good now!
